src/api/db/student_course_selection.py

```diff
@@ -0,0 +1,72 @@
+class student_course_selection:
+	def __init__(self, db_conn):
+		self.db_conn = db_conn
+
+	def add_selection(self, cid, sem, uid):
+		try:
```

Comment:
You shouldn't have to wrap these queries in a `try` block. The `db_conn` is already a wrapper around the psycopg2 driver which will do the actual DB query call in a `try` block and return the corresponding error on exception like you have done here.